codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ConstrainedNumberGenerator.kt

```diff
@@ -57,6 +59,30 @@ class ConstrainedShortGenerator(
     }
 }
 
+class ConstrainedLongGenerator(
+    val codegenContext: ServerCodegenContext,
+    val writer: RustWriter,
+    val shape: LongShape,
+) {
+    val inner = ConstrainedNumberGenerator(codegenContext, writer, shape, RustType.Integer(64))
+
+    fun render() {
+        inner.render()
+    }
+}
+
+class ConstrainedByteGenerator(
+    val codegenContext: ServerCodegenContext,
+    val writer: RustWriter,
+    val shape: ByteShape,
+) {
+    val inner = ConstrainedNumberGenerator(codegenContext, writer, shape, RustType.Integer(8))
+
+    fun render() {
+        inner.render()
+    }
+}
```

Comment:
It feels like these 4 could be DRY'd up a bit more, but all I can see is using inheritance, which I'm not a fan of.

Reply:
Lovely